tools/ci_build/build.py

```diff
@@ -188,6 +188,8 @@ def use_dev_mode(args):
         return False
     if is_macOS() and (args.ios or args.visionos or args.tvos):
         return False
+    if args.use_qnn:
```

Comment:
did we confirm this is needed because of the check (lines 193-194) below?
why is it needed currently? would we want to get rid of it? 
are there other pipelines that would fail if we got rid of it